src/rules/noDuplicateVariableRule.ts

```diff
@@ -27,6 +27,27 @@ class NoDuplicateVariableWalker extends Lint.ScopeAwareRuleWalker<ScopeInfo> {
         return new ScopeInfo();
     }
 
+    fail(ident: ts.Identifier): void {
```

Comment:
Can you please name this something slightly more descriptive (addFailureOnIdentifier or something like that). Additionally, this should be marked as private and moved to the end of the class (since it's a private function).